src/preattestation.rs

```diff
@@ -305,11 +306,16 @@ mod idblock {
     pub fn generate_id_block(args: Args, quiet: bool) -> Result<()> {
         let ld =
             if &args.launch_digest[..args.launch_digest.char_indices().nth(2).unwrap().0] == "0x" {
-                SnpLaunchDigest::new(Vec::from_hex(&args.launch_digest[2..])?.try_into()?)
+                SnpLaunchDigest::new(
+                    Vec::from_hex(&args.launch_digest[2..])?
+                        .as_slice()
+                        .try_into()?,
+                )
             } else {
                 SnpLaunchDigest::new(
                     general_purpose::STANDARD
                         .decode(args.launch_digest)?
+                        .as_slice()
                         .try_into()?,
                 )
```

Comment:
Gets rid of a try_into
```suggestion
                SnpLaunchDigest::from_bytes(
                    general_purpose::STANDARD
                        .decode(args.launch_digest)?
                        .as_slice(),
                )?
```